GetandTake/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Security.Claims;
+using System.Text;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Identity.UI.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.WebUtilities;
+
+namespace GetandTake.Areas.Identity.Pages.Account;
+
+[AllowAnonymous]
+public class ExternalLoginModel : PageModel
+{
+    private readonly SignInManager<IdentityUser> _signInManager;
+    private readonly UserManager<IdentityUser> _userManager;
+    private readonly IUserStore<IdentityUser> _userStore;
+    private readonly IUserEmailStore<IdentityUser> _emailStore;
+    private readonly IEmailSender _emailSender;
+    private readonly ILogger<ExternalLoginModel> _logger;
+
+    public ExternalLoginModel(
+        SignInManager<IdentityUser> signInManager,
+        UserManager<IdentityUser> userManager,
+        IUserStore<IdentityUser> userStore,
+        ILogger<ExternalLoginModel> logger,
+        IEmailSender emailSender)
+    {
+        _signInManager = signInManager;
+        _userManager = userManager;
+        _userStore = userStore;
+        _emailStore = GetEmailStore();
+        _logger = logger;
+        _emailSender = emailSender;
+    }
+
+    [BindProperty]
+    public InputModel Input { get; set; }
+
+    public string ProviderDisplayName { get; set; }
+
+    public string ReturnUrl { get; set; }
+
+    [TempData]
+    public string ErrorMessage { get; set; }
+
+    public class InputModel
```

Comment:
Just an advice: inner classes should be declared before public properties, so, let's put it between constructor and first public property (add updates for other similar cases, please)